library.lua

```diff
@@ -103,17 +103,18 @@ function Library:showDirectoryView(path_to_directory)
             end
       })
    end
-   UIManager:show(KeyValuePage:new{
-         title = "Puzzles",
-         kv_pairs = kv_pairs,
-		 close_callback = function() self:onClose() end
-   })
+   local LibraryPage = KeyValuePage:new{
+      title = "Puzzles",
+      kv_pairs = kv_pairs,
+      close_callback = function() self:onClose() end,
+   }
+   UIManager:show(LibraryPage)
 end
 
 function Library:onClose()
-	UIManager:restartKOReader()
-	--UIManager:close(self)
-	return true
+   UIManager:restartKOReader()
```

Comment:
I don't think we need to restart KOReader just to refresh the display. Your use of `Screen`'s `clear` and `refreshFull` should work?